integration-tests/google-storage/src/main/java/org/apache/camel/quarkus/component/google/storage/it/GoogleStorageResource.java

```diff
@@ -0,0 +1,172 @@
+            String port = System.getProperty(GoogleStorageResource.PARAM_PORT);
+            gsc.getConfiguration().setStorageClient(StorageOptions.newBuilder()
+                    .setHost("http://localhost:" + port)
+                    .setProjectId("dummy-project-for-testing")
+                    .build()
+                    .getService());
+        } else {
+            gsc.getConfiguration().setStorageClient(sp.storage());
+        }
+        return gsc;
+    }
+
+    @Path("/operation")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    public String operation(Map<String, Object> parameters,
+            @QueryParam(QUERY_OPERATION) String operation,
+            @QueryParam(QUERY_BUCKET) String bucketName) throws Exception {
+        GoogleCloudStorageOperations op = GoogleCloudStorageOperations.valueOf(operation);
+        String url = getBaseUrl(bucketName, "operation=" + op.toString());
+        final Object response = producerTemplate.requestBodyAndHeaders(url, null, parameters, Object.class);
+        if (response instanceof Blob) {
+            return new String(((Blob) response).getContent());
+        }
+        if (response instanceof CopyWriter) {
+            return new String(((CopyWriter) response).getResult().getContent());
+        }
+        if (response instanceof List) {
+            List l = (List) response;
+            return (String) l.stream().map(o -> {
+                if (o instanceof Bucket) {
+                    return ((Bucket) o).getName();
+                }
+                if (o instanceof Blob) {
+                    return ((Blob) o).getName();
+                }
+                return "null";
+            }).collect(Collectors.joining(","));
+        }
+        return String.valueOf(response);
+    }
+
+    @Path("/putObject")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response putObject(String body,
+            @QueryParam(QUERY_BUCKET) String bucketName,
+            @QueryParam(QUERY_OBJECT_NAME) String objectName) throws Exception {
+        String url = getBaseUrl(bucketName, "autoCreateBucket=true");
+        final Blob response = producerTemplate.requestBodyAndHeader(url,
+                body,
+                GoogleCloudStorageConstants.OBJECT_NAME, objectName, Blob.class);
+        return Response
+                .created(new URI("https://camel.apache.org/"))
+                .entity(response.getName())
+                .build();
+    }
+
+    @Path("/startPolling")
+    @POST
+    public void startPolling(@QueryParam(QUERY_BUCKET) String bucketName,
+            @QueryParam(QUERY_POLLING_ACTION) String pollingAction,
+            @QueryParam(QUERY_DESTINATION_BUCKET) String destinationBucket) {
+        // use another thread for polling consumer to demonstrate that we can wait before
+        // the message is sent to the queue
+        Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
For this kind of thing, it's far easier to create a `RouteBuilder`, configure the `google-storage` consumer and direct any messages to a `seda` or `mock` endpoint which you can then retrieve via a `ConsumerTemplate` in the JAX-RS resource class.

It makes the test flow much better...

1. Create some storage object
2. Retrieve storage object

There's no need for executors, polling etc.

Reply:
No problem with it. I thought that not using seda|mock|direct libraries is better solution. I'll use route then.

Reply:
I used a route builder to poll into direct and then receive message from it.